pkg/scheduler/internal/cache/cache.go

```diff
@@ -221,7 +229,22 @@ func (cache *schedulerCache) UpdateNodeInfoSnapshot(nodeSnapshot *nodeinfosnapsh
 			node.info.TransientInfo.ResetTransientSchedulerInfo()
 		}
 		if np := node.info.Node(); np != nil {
-			nodeSnapshot.NodeInfoMap[np.Name] = node.info.Clone()
+			existing, ok := nodeSnapshot.NodeInfoMap[np.Name]
+			if !ok {
+				updateAllLists = true
+				existing = schedulernodeinfo.NewNodeInfo()
+				nodeSnapshot.NodeInfoMap[np.Name] = existing
+			}
+			new := node.info.Clone()
+			// We track nodes that have pods with affinity, here we check if this node changed its
+			// status from having pods with affinity to NOT having pods with affinity or the other
+			// way around.
+			if (len(existing.PodsWithAffinity()) > 0) != (len(new.PodsWithAffinity()) > 0) {
+				updateNodesHavePodsWithAffinity = true
+			}
+			// We need to preserve the original pointer of the NodeInfo struct since it
+			// is used in the NodeInfoList.
+			*nodeSnapshot.NodeInfoMap[np.Name] = *new
```

Comment:
I think `*existing=*new` is more readable.

Reply:
sg